airflow/configuration.py

```diff
@@ -271,7 +271,7 @@ def __init__(self, default_config=None, *args, **kwargs):
         if default_config is not None:
             self.airflow_defaults.read_string(default_config)
             # Set the upgrade value based on the current loaded default
-            default = self.airflow_defaults.get('logging', 'log_filename_template', fallback=None, raw=True)
+            default = self.airflow_defaults.get('logging', 'log_filename_template', fallback=None)
```

Comment:
Since we now set an env var, we want the `parsed` value, not the `raw` value. Otherwise we end up with `%%`'s instead of the `%`'s that jinja is expecting.

I feel like there might be more landmines in this are (configparser vs env var when shoving these back into env vars always), but this at least fixes this case.